scenarios/erc20_multichain_factory/test/DynamicContractRecovery_test.res

```diff
@@ -253,19 +253,51 @@ describe("Dynamic contract restart resistance test", () => {
           ->Array.flatMap(set => set->Belt.Set.String.toArray)
 
         Assert.deepEqual(
-          [Mock.mockDyamicToken1->Address.toString, Mock.mockDyamicToken2->Address.toString],
           restartedChainFetcher.dynamicContractPreRegistration->Option.getExn->Js.Dict.keys,
+          [Mock.mockDynamicToken1->Address.toString, Mock.mockDynamicToken2->Address.toString],
           ~message="Should return all the dynamic contracts related to handler that uses preRegistration",
         )
 
         Assert.deepEqual(
-          [],
           dynamicContracts,
+          [],
           ~message="Should have no dynamic contracts yet since this tests the case starting in preregistration",
         )
         resetEventOptionsToOriginal()
       }
 
+      Assert.deepEqual(
+        stubDataInitial->Stubs.getTasks,
+        [
+          NextQuery(CheckAllChains),
+          Mock.getUpdateEndofBlockRangeScannedData(
+            Mock.mockChainDataMap,
+            ~chain=Mock.Chain1.chain,
+            ~blockNumberThreshold=-197,
+            ~blockTimestampThreshold=25,
+            ~blockNumber=3,
+          ),
+          UpdateChainMetaDataAndCheckForExit(NoExit),
+          ProcessEventBatch,
+          NextQuery(Chain(Mock.Chain1.chain)),
+          NextQuery(Chain(Mock.Chain2.chain)),
+          UpdateChainMetaDataAndCheckForExit(NoExit),
+          ProcessEventBatch,
+          NextQuery(CheckAllChains),
+        ],
+        ~message="This looks wrong, but snapshot to track how it changes with time",
+      )
+      // DynamicContract
+      // fromBlock: 0
+      // toBlock: 0
+      await dispatchAllTasks()
+      // DynamicContract
+      // fromBlock: 0
+      // toBlock: 3
+      await dispatchAllTasks()
+      // DynamicContract
+      // fromBlock: 2
+      // toBlock: 3
       await dispatchAllTasks()
```

Comment:
Previously all 3 queries of the same fetchState somehow managed to run during a single `await dispatchAllTasks()` call. Now it the run one by one, thanks to the mutable fetching partitions set.